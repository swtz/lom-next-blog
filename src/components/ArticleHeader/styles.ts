import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      font-size: ${theme.font.sizes.xxlarge};
      margin: 0;
      margin-bottom: ${theme.spacings.medium};

      @media ${theme.media.lteMedium} {
        font-size: ${theme.font.sizes.large};
      }
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    margin: ${theme.spacings.medium} 0;

    @media ${theme.media.lteMedium} {
      font-size: 2rem;
    }
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
