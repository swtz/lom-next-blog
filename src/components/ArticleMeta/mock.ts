import { ArticleMetaProps } from '.';

export default {
  createdAt: '2025-02-12T11:41:13.991Z', // value from current Strapi API
  author: {
    data: {
      id: '1',
      attributes: {
        displayName: 'surfista',
        slug: 'surfista',
      },
    },
  },

  categories: {
    data: [
      {
        id: '4',
        attributes: {
          displayName: 'Off Topic',
          slug: 'off-topic',
        },
      },
      {
        id: '1',
        attributes: {
          displayName: 'tech',
          slug: 'tech',
        },
      },
    ],
  },
} as ArticleMetaProps;
