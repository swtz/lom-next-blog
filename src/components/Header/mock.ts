import { HeaderProps } from '.';

export default {
  blogName: 'Blog do Arroz Ã© vida!',
  blogDescription: `Rice is a cereal grain and in its domesticated form is the staple food of over half of the world's population, particularly in Asia and Africa. Rice is the seed of the grass species Oryza sativa.`,
  logo: '/assets/images/logo.svg',
  showText: true,
} as HeaderProps;
