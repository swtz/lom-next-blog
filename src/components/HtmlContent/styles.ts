import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-size: calc(${theme.font.sizes.xsmall} + 0.2rem);
    line-height: 1.5;

    p {
      margin: ${theme.spacings.medium} 0;
    }

    a,
    a:visited,
    a:link {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;
    }

    a:hover {
      filter: brightness(50%);
    }

    code {
      font-family: monospace;
      color: ${theme.colors.secondary};
      font-size: ${theme.font.sizes.xsmall};
      background: ${theme.colors.mediumGray};
      padding: 0.2rem;
      margin: 0.2rem;
    }

    pre {
      background: ${theme.colors.primary};
      padding: ${theme.spacings.medium};
      font-family: monospace;
      color: ${theme.colors.white};
      margin: ${theme.spacings.medium} 0;
      font-size: ${theme.font.sizes.xsmall};
      width: 100%;
      overflow-x: auto;
    }

    pre code {
      color: inherit;
      background: inherit;
    }

    img {
      max-width: 100%;
    }

    .image {
      background: ${theme.colors.mediumGray};
      line-height: 0;
      margin: ${theme.spacings.medium} 0;
    }

    .image figcaption {
      background: ${theme.colors.mediumGray};
      font-size: ${theme.font.sizes.xsmall};
      padding: ${theme.spacings.small};
      text-align: center;
      line-height: 1.3;
    }

    .image-style-side {
      float: right;
      max-width: 50%;
      margin: ${theme.spacings.medium} ${theme.spacings.small};
    }

    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.mediumGray};
    }

    ul,
    ol {
      margin: ${theme.spacings.medium} ${theme.spacings.xlarge};
    }

    .table {
      width: 100%;
      overflow: hidden;
      overflow-x: auto;
    }

    table {
      width: 100%;
      border-collapse: collapse;
      margin: ${theme.spacings.medium} 0;
    }

    table td,
    table th {
      padding: ${theme.spacings.small};
      border: 0.1rem solid ${theme.colors.mediumGray};
    }

    .media iframe {
      width: 100%;
      height: auto;
      aspect-ratio: 16 / 9;
    }

    blockquote {
      border-left: 0.5rem solid ${theme.colors.secondary};
      color: ${theme.colors.darkerGray};
      filter: brightness(80%);
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      margin: ${theme.spacings.medium};
    }

    @media ${theme.media.lteMedium} {
      font-size: 2rem;

      pre {
        font-size: 1.2rem;
      }

      .image-style-side {
        float: none;
        max-width: 100%;
        margin: ${theme.spacings.xlarge} 0;
      }
    }
  `};
`;
