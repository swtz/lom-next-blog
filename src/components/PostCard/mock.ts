import { data } from '../../api/dados.json';

export const postId = data.posts.data[0].id;
export const postProps = data.posts.data[0].attributes;
