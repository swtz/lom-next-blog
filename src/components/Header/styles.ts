import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    /* padding: ${theme.spacings.xhuge}; */
    width: 100%;
    max-width: ${theme.sizes.max};
    color: ${theme.colors.darkText};
    font-size: ${theme.font.sizes.xsmall};
    margin: 0 auto;

    ${HeadingStyles} {
      margin: 0 0 calc(${theme.spacings.xsmall} - 1rem);
    }

    @media ${theme.media.lteXSmall} {
      flex-flow: column wrap;

      > div {
        ${HeadingStyles} {
          margin: ${theme.spacings.xsmall} 0
            calc(${theme.spacings.small} - 1rem);
        }
      }
    }
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;
  `}
`;
