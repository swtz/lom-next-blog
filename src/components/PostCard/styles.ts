import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.div`
  text-align: center;

  ${({ theme }) => css`
    ${HeadingStyles} {
      margin: 0;
      margin-top: ${theme.spacings.small};
      margin-bottom: ${theme.spacings.small};
      transition: all 300ms ease-in-out;
    }

    a {
      text-decoration: none;
    }

    &:hover ${HeadingStyles} {
      color: ${theme.colors.secondary};
    }

    &:hover img {
      opacity: 0.8;
    }
  `}
`;

export const Cover = styled.img`
  ${() => css`
    max-width: 100%;
    transition: opacity 300ms ease-in-out;
  `}
`;
export const Excerpt = styled.p`
  ${() => css``}
`;
