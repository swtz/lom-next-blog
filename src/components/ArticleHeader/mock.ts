import { ArticleHeaderProps } from '.';
import { ArticleMetaProps } from '../ArticleMeta';

const metadata = {
  createdAt: '2025-02-12T11:41:13.991Z',
  author: {
    data: {
      id: '1',
      attributes: {
        displayName: 'surfista',
        slug: 'surfista',
      },
    },
  },

  categories: {
    data: [
      {
        id: '4',
        attributes: {
          displayName: 'Off Topic',
          slug: 'off-topic',
        },
      },
      {
        id: '1',
        attributes: {
          displayName: 'tech',
          slug: 'tech',
        },
      },
    ],
  },
} as ArticleMetaProps;

export default {
  title: 'Hello, rice! With rice is possible make a several delicious dishes',
  excerpt: `Rice is a cereal grain and in its domesticated form is the staple food of over half of the world's population, particularly in Asia and Africa. Rice is the seed of the grass species Oryza sativa (Asian rice).`,
  cover: {
    data: {
      id: 1,
      attributes: {
        url: '/assets/images/arroz-branco-saudavel.webp',
        alternativeText: null,
      },
    },
  },
  metadata: metadata,
} as ArticleHeaderProps;
