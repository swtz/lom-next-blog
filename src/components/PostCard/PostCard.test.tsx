import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { PostCard, PostCardProps } from '.';
import { postProps } from './mock';

const props: PostCardProps = postProps;

describe('<PostCard />', () => {
  it('should render a heading, cover and except', () => {
    renderTheme(<PostCard {...props} />);

    expect(
      screen.getByRole('heading', { name: props.title }),
    ).toBeInTheDocument();

    expect(
      screen.getByAltText(
        'a fox holding a box containing a bottle and ice cubes',
      ),
    ).toBeInTheDocument();

    expect(screen.getByText(props.excerpt)).toBeInTheDocument();

    expect(screen.getAllByRole('link')[0]).toHaveAttribute(
      'href',
      `/post/${props.slug}`,
    );
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<PostCard {...props} />);

    expect(container).toMatchSnapshot();
  });
});
