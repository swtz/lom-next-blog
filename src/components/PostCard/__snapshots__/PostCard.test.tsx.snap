// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
.c3 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: none;
}

.c0 {
  text-align: center;
}

.c0 .c2 {
  margin: 0;
  margin-top: .8rem;
  margin-bottom: .8rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover .c2 {
  color: #DC143C;
}

.c0:hover img {
  opacity: 0.8;
}

.c1 {
  max-width: 100%;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
}

<div>
  <div
    class="c0"
  >
    <a
      href="/post/hello-fox"
    >
      <img
        alt="a fox holding a box containing a bottle and ice cubes"
        class="c1"
        src="https://pub-aea127bc483c40dea7b9339692296909.r2.dev/DALL_E_2024_10_26_12_58_20_A_modern_logo_design_for_Deck_da_Praia_Conveniencia_featuring_a_fox_mascot_representing_cold_beer_delivery_The_fox_should_appear_in_motion_with_sp_fd834176b6.webp"
      />
    </a>
    <a
      href="/post/hello-fox"
    >
      <h2
        class="c2 c3"
      >
        Hello, fox!
      </h2>
    </a>
    <p
      class=""
    >
      Foxes are small-to-medium-sized omnivorous mammals belonging to several genera of the family Canidae.
    </p>
  </div>
</div>
`;
